app/components/schema-field-text/component.js

```diff
@@ -0,0 +1,21 @@
+import Ember from 'ember';
+
+export default Ember.Component.extend({
+  init() {
+    this._super(...arguments);
+
+    let key = this.get('key');
+    let document = this.get('document');
+    let defaultValue = this.get('property.default');
+    let initialValue = document.get(key) || defaultValue || '';
+
+    this.set('value', initialValue);
+  },
+
+  setValue: Ember.observer('value', function() {
```

Comment:
Lets pair on removing this observer.

We can use the one-way-input we made for dashboard that triggers an action upon change (instead of implicit two-way bindings), that would let us avoid the observer completely. @poteto extracted that one-way-input into an addon (https://github.com/dockyard/ember-one-way-input) so it should be easy to pull in here.